content/blog/2024/10/what-is-the-esp-registry/index.md

```diff
@@ -0,0 +1,160 @@
+---
+title: "What is the ESP Component Registry?"
+date: 2024-10-27
+showAuthor: false
+authors:
+  - "pedro-minatel"
+tags: ["Registry", "Component", "ESP-IDF"]
+---
+
+When starting a new project, we often reuse existing code, for example from libraries and drivers. This not only saves time but also makes your project easier to maintain.
+
+In this article, we will explore what the [ESP Component Registry](https://components.espressif.com) is and the key concepts of components, making it easy to understand not only how to use components but also how to contribute.
+
+## Concept
+
+The architecture of [ESP-IDF](https://idf.espressif.com) is modular, making it easier to understand and maintain. Such modules are called *components*, as opposed to *libraries*, because they are self-contained and can also communicates with other components. Components are used on top of the ESP-IDF core.
+
+There are three concepts that we need to introduce in order to understand the complete solution: component, component manager, and component register. Let's go through them one by one.
+
+### 1. Component
+
+As mentioned above, a component is a self-contained piece of code that contains new functionality and can be added to an application. In most cases, components are used in multiple applications, making it easy to reuse code, improve the efficiency, and reduce development complexity.
+
+For example, let's imagine there is an application called `my-application` in which we want to use a specific sensor that will collect temperature and humidity from the environment. Instead of adding all the logic behind the communication protocol, data collection, conversions, etc. to the application code, you can add a component i.e `sensor-component`, that has this logic already implemented in code. The important part is that this code has been reviewed and tested multiple times by other developers.
+
+The component can be created directly in the application, inside a specific folder, named `components` or can be created in a separated component project and this component can be added as a dependency to your application. Both solutions works but the recommended way is to create a new component project and add it as a dependency.

```

Comment:
```suggestion
The component can be either created directly in the application, inside a specific folder named `components`, or it can be created in a separated component project and then added as a dependency to the application. Both solutions work but the recommended way is to create a new component project and add it as a dependency.
```